src/Export/CsvTarget.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * @file
+ * Contains CSV implementation of ExtractionTargetInterface.
+ */
+
+namespace App\Export;
+
+use App\Document\Entry;
+use App\Document\ExtractionResult;
+use Box\Spout\Writer\Common\Creator\WriterEntityFactory;
+
+/**
+ * Class CsvTarget.
+ */
+class CsvTarget implements ExtractionTargetInterface
+{
+    /* @var \Box\Spout\Writer\WriterInterface $writer */
+    private $writer;
+    /* @var string $filename */
+    private $filename;
+    /* @var array $types */
+    private $types = null;
+
+    /**
+     * CsvTarget constructor.
+     *
+     * @param string $filename
+     */
+    public function __construct(string $filename)
+    {
+        $this->filename = $filename;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function initialize(): void
+    {
+        $this->writer = WriterEntityFactory::createCSVWriter();
+
+        $this->writer->openToFile($this->filename);
+
+        // Add headers.
+        $headers = [
+            'elasticId',
+            'date',
+            'clientId',
+            'agency',
+            'event',
+            'identifierType',
+            'materialId',
+            'response',
+            'imageId',
+        ];
+        $rowFromValues = WriterEntityFactory::createRowFromArray($headers);
+        $this->writer->addRow($rowFromValues);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function finish(): void
+    {
+        $this->writer->close();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addEntry(Entry $entry): void
+    {
+        $values = [
+            $entry->getElasticId(),
+            $entry->getDate()->format('c'),
+            $entry->getClientId(),
+            $entry->getAgency(),
+            $entry->getEvent(),
+            $entry->getIdentifierType(),
+            $entry->getMaterialId(),
+            $entry->getResponse(),
+            $entry->getImageId(),
+        ];
+        $rowFromValues = WriterEntityFactory::createRowFromArray($values);
+        $this->writer->addRow($rowFromValues);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function entryExists($id): bool
+    {
+        // Since this is a spreadsheet export, we are not concerned with already added entries.
+        return false;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function recordExtractionResult(ExtractionResult $extractionResult): void
+    {
```

Comment:
Comment about why this is empty would be [grrreat](https://upload.wikimedia.org/wikipedia/en/thumb/c/c8/Tony_the_Tiger_%28Kellogg%27s_Frosted_Flakes%27_mascot%29.jpg/220px-Tony_the_Tiger_%28Kellogg%27s_Frosted_Flakes%27_mascot%29.jpg) (tony)

Reply:
added